sycl/include/sycl/ext/oneapi/properties/property.hpp

```diff
@@ -266,6 +266,8 @@ template <typename PropertyT> struct PropertyMetaInfo {
   static constexpr std::nullptr_t value = nullptr;
 };
 
+template <typename> struct HasCompileTimeEffect : std::false_type {};
```

Comment:
I think it's a bad decision to extend generic properties interface (even the internal one) with some trait specific to just a few properties/APIs accepting them.